generator.yaml

```diff
@@ -121,17 +154,25 @@ operations:
     set_output_custom_method_name: CustomCreateCacheParameterGroupSetOutput
   DescribeCacheParameterGroups:
     set_output_custom_method_name: CustomDescribeCacheParameterGroupsSetOutput
+  CreateUser:
+    set_output_custom_method_name: CustomCreateUserSetOutput
+  ModifyUser:
+    custom_implementation: CustomModifyUser
+    set_output_custom_method_name: CustomModifyUserSetOutput
```

Comment:
These can be removed and required logic can be added to `hooks` configuration that is already specified for these methods in this PR.

Reply:
We need to do 2 things after receiving an API response:

1. set the synced condition based on the User's `status.status`
2. update the state-keeping `status` fields of `LastRequestedAccessString` and `ExpandedAccessString`, but only upon successful create and modify calls. The logic to handle the access string depends on `LastRequestedAccessString` to not be changed upon describe calls.

It's certainly possible to move the first item to the custom hooks, but then we would need two separate hook functions:
1. One that gets called from create and modify, that does both the items above.
2. One that gets called from read, that does only the second item above.

I think this approach might make first hook a bit too convoluted in terms of combining two tasks into one. Further, since it would be called from both `sdkCreate` and `sdkUpdate`, but the API response types are different, we would need to pass in both `resp.Status` and `resp.AccessString` as separate arguments. 

I chose the initial approach since the task accomplished by the hook and the task accomplished by the `set_output` functions are separated and well-defined, but if I'm missing a simpler way to do this please let me know.

Reply:
thanks for the details @echen-98 
lets keep it as for now